### DISCUSSION TITLE
How to detect VCP pattern

### DISCUSSION THREAD

**Discussion by sunel** [pranjal-joshi/Screeni-py#97] [General]
Latest one is KHAICHEM https://in.tradingview.com/chart/?symbol=NSE%3AKHAICHEM

![image](https://user-images.githubusercontent.com/1009777/150180684-c8816d80-c916-410e-bbe2-ce2446fe7868.png)

**Comment by pranjal-joshi** [comment]
Hello @sunel

I have a pseudo-code of this feature since long time.
I have plans to implement, test and integrate in upcoming versions.
If you're willing to add this feature, I will welcome your code contribution.

Do let me know if it's possible.

**Reply by sunel** [reply]
i am totally new to python, i am understanding the code. i saw that you have added card in the project section https://github.com/pranjal-joshi/Screeni-py/projects/1#card-60583736

Will try to contribute.

**Reply by pranjal-joshi** [reply]
Hello,

The project card you've mentioned is the logic that I think will work.

Before that, Earlier I tried implementing VCP with another approach which wasn't working as expected.

You can try to understand the code [here](https://github.com/pranjal-joshi/Screeni-py/blob/main/src/classes/Screener.py#L434), because atleast the data preprocessing part will be similar, only condition checking needs to be modified.

**Reply by sunel** [reply]
Sure will do 👍

**Reply by pranjal-joshi** [reply]
I've tried writing [this](https://github.com/pranjal-joshi/Screeni-py/blob/ec0114cbaf580cab2e2213b561437c8e97a75b87/src/classes/Screener.py#L433) method in ec0114c.
It is barely working and not much accurate. Due to the variable timespan of contractions, the results obtained aren't yet useful.
Let's see if you can come up with some solution for this, please [contribute](https://github.com/pranjal-joshi/Screeni-py/blob/new-features/CONTRIBUTING.md).

**Reply by sunel** [reply]
Awesome will test this and try to improve it

**Reply by pranjal-joshi** [reply]
There is no other option for VCP so far.
Use Option 2 (Break-out) for time being
